### ISSUE TITLE
[Feature Request] Export/Import Block List

### ISSUE THREAD

**Issue by 100chilly** [vrcx-team/VRCX#927]
**Explain in detail what your suggested feature would be used for.**
A simple export/import of your block list in different file formats.

**Describe how it would look if it requires a UI.**
N/A

**Explain why people would want to use it.**
For people who want a fresh start without the bullshit of drama starters, problematic people, etc. This is an easy way to just transfer from one account to another.

**Comment by DubyaDude** [comment]
Bulk moderation is a bit tricky, and from my foggy memory about it, is something VRChat isn't fond of. From what was discussed in the discord, it being something that can affect someone's trust ranking, also doesn't help.

**Comment by plautus2** [comment]
PLEASE ADD THIS! IT IS SO IMPORTANT!

**Comment by plautus2** [comment]
omg, o h my god. Most needed feature, they refuse to add. THis is laughable

**Comment by Myrkie** [comment]
we are not inclined to add features you request, VRCX is a free application, if you want this feature add it yourself.

**Comment by plautus2** [comment]
sure, I go spent hundrets of hours into learning how to program just so I can add this block/mute list ex/importer, even thought u could just add it in a couple minutes. sure....

**Comment by ScruffyRules** [comment]
> omg, o h my god. Most needed feature, they refuse to add. THis is laughable

who are you

**Comment by plautus2** [comment]
at this point I think it's just fear of the vrx developers. They are scared they program might be moderated or something by the vrchat team. Absoloute  cowardish

**Comment by SalbugVR** [comment]
> sure, I go spent hundrets of hours into learning how to program just so I can add this block/mute list ex/importer, even thought u could just add it in a couple minutes. sure....

Here you go, have fun.

[app.js](https://github.com/vrcx-team/VRCX/blob/master/html/src/app.js)

**Comment by GroovyTeacup** [comment]
> omg, o h my god. Most needed feature, they refuse to add. THis is laughable

![lol](https://i.imgur.com/IAl2IyN.gif)

**Comment by plautus2** [comment]
u guys should retire, if u cant even make a block/mute list import/export mechanic

**Comment by Myrkie** [comment]
> at this point I think it's just fear of the vrx developers. They are scared their program might be moderated or something by the vrchat team. Absoloute cowardish. I doubt they ever do anything daring!

Hi, VRCX team has direct communication to VRChat's staff they routinely talk with us since VRCX has reached a large amount of users. this is not assumptions VRChat has asked us in the past to make changes to VRCX and expressed displeasure in adding the feature automated moderation, we are not adding the feature for this reason.

**Comment by plautus2** [comment]
communicate with them again, stop being a coward. VRchat staff must think they got u on the balls at this point

**Comment by SalbugVR** [comment]
> u guys should retire, if u cant even make a block/mute list import/export mechanic

You’re welcome to fork the VRCX project and implement this feature yourself if you’d like. However, the maintainers have decided not to include this feature, as it can be easily abused and may put its users at risk of being flagged for moderation. Thank you for understanding.

**Comment by plautus2** [comment]
I'm speechless.

**Comment by OctoFloofy** [comment]
I hope you're just a troll at this point

**Comment by plautus2** [comment]
> From what was discussed in the discord, it being something that can affect someone's trust ranking, also doesn't help.

This is a complete invalid statement, since they can simply add a warning message to the use of this feature, so everyone can decide for himself if he wants to make use of it or not.

**Comment by SalbugVR** [comment]
> > From what was discussed in the discord, it being something that can affect someone's trust ranking, also doesn't help.
> 
> so everyone can decide for himself if he wants to make use of it or not.

It's not just about an individual deciding whether they want to use it. The main concern is that simply having an abusable mass-moderation system within the program would be enough of a red flag for VRChat to take action against VRCX and its users. Either way, VRChat would likely instruct the maintainers to remove it, and they would, of course, comply. We’re sorry, but we don’t intend to drag the application's users down due to a lack of foresight on your part.

Again, if you want this functionality, consider learning to code, forking the project, and implementing it yourself.

**Comment by plautus2** [comment]
how is this abuse? I am not talking to mass block all users in a room, im talking about transfering my own block lists. Nothiung u've staied or these other guys said remotely make any sense.

**Comment by Myrkie** [comment]
> how is this abuse? I am not talking to mass block all users in a room, im talking about transfering my own block lists. Nothiung u've staied or these other guys said remotely make any sense.

applying moderation's against any users affects their rank such as blocking and muting, VRChat is very much against this and any sort of automated moderation's can be abused or used in a targeted manor (KOS Lists) thus why VRChat has expressed a dislike in the idea in the past.

**Comment by SalbugVR** [comment]
> how is this abuse? I am not talking to mass block all users in a room, im talking about transfering my own block lists. Nothiung u've staied or these other guys said remotely make any sense.

The original post states.

> Explain in detail what your suggested feature would be used for.
A simple export/import of your block list in different file formats.

Keyword being 'file formats'. I really shouldn't have to spell this out for you.

**Comment by regalialong** [comment]
> how is this abuse? I am not talking to mass block all users in a room, im talking about transfering my own block lists. Nothiung u've staied or these other guys said remotely make any sense.

VRCX can't just transfer the entirety of your blocks, it has to re-do every block individually one by one, which is spammy especially if you have a lot of blocks. :p

**Comment by SalbugVR** [comment]
To put it simply for you

> how is this abuse? I am not talking to mass block all users in a room, im talking about transfering my own block lists. Nothiung u've staied or these other guys said remotely make any sense.

Simply put, users could create KOS lists from this if the blocks and mutes lists were made exportable and importable as file formats. This would also place a heavy load on their servers due to the high volume of API calls resulting from user moderation actions occurring simultaneously.

**Comment by plautus2** [comment]
> > how is this abuse? I am not talking to mass block all users in a room, im talking about transfering my own block lists. Nothiung u've staied or these other guys said remotely make any sense.
> 
> applying moderation's against any users affects their rank such as blocking and muting, VRChat is very much against this and any sort of automated moderation's can be abused or used in a targeted manor (KOS Lists) thus why VRChat has expressed a dislike in the idea in the past.

I could simply reblock everyone, that would also affect their rank. But why when it can be achieved in a better way. If I really wanna hurt a user, I make 100 accounts, and block on all of these accounts a certain user. 
What u say makes 0 sense again.

**Comment by plautus2** [comment]
> To put it simply for you
> 
> > how is this abuse? I am not talking to mass block all users in a room, im talking about transfering my own block lists. Nothiung u've staied or these other guys said remotely make any sense.
> 
> Simply put, users could create KOS lists from this if the blocks and mutes lists were made exportable and importable as file formats. This would also place a heavy load on their servers due to the high volume of API calls resulting from user moderation actions occurring simultaneously.

the issue ur describing here isnt an importer tool, but creating mass accounts

**Comment by Turretoforth** [comment]
> > To put it simply for you
> > > how is this abuse? I am not talking to mass block all users in a room, im talking about transfering my own block lists. Nothiung u've staied or these other guys said remotely make any sense.
> > 
> > 
> > Simply put, users could create KOS lists from this if the blocks and mutes lists were made exportable and importable as file formats. This would also place a heavy load on their servers due to the high volume of API calls resulting from user moderation actions occurring simultaneously.
> 
> this isnt even remotely possible

It's simple math really.
Easier to use = more usage
A single block from you is nothing, but a block list circulating is potentially several hundred of users blocking the same person

**Comment by plautus2** [comment]
let them block.  Is their loss

**Comment by regalialong** [comment]
For what it's worth, VRCX also already has a high support volume. Giving users a footgun will eventually cause users to trickle in that managed to shoot themselves, it's practically in the best interest of nobody.

**Comment by SalbugVR** [comment]
> > To put it simply for you
> > > how is this abuse? I am not talking to mass block all users in a room, im talking about transfering my own block lists. Nothiung u've staied or these other guys said remotely make any sense.
> > 
> > 
> > Simply put, users could create KOS lists from this if the blocks and mutes lists were made exportable and importable as file formats. This would also place a heavy load on their servers due to the high volume of API calls resulting from user moderation actions occurring simultaneously.
> 
> this isnt even remotely possible

In your other responses, you mention that you don’t understand and that it makes no sense to you, yet you’re telling me this isn’t even remotely possible? So you're telling me you do understand.

**Comment by plautus2** [comment]
> > > To put it simply for you
> > > > how is this abuse? I am not talking to mass block all users in a room, im talking about transfering my own block lists. Nothiung u've staied or these other guys said remotely make any sense.
> > > 
> > > 
> > > Simply put, users could create KOS lists from this if the blocks and mutes lists were made exportable and importable as file formats. This would also place a heavy load on their servers due to the high volume of API calls resulting from user moderation actions occurring simultaneously.
> > 
> > 
> > this isnt even remotely possible
> 
> In your other responses, you mention that you don’t understand and that it makes no sense to you, yet you’re telling me this isn’t even remotely possible? So you're telling me you do understand.

No point creating KOS list, when u can make one of ur own and import

**Comment by Myrkie** [comment]
> No point creating KOS list, when u can make one of ur own and import


so you can implement the feature yourself? great, looking forward to the pull request. cheers!

**Comment by plautus2** [comment]
I know u guys can

**Comment by GroovyTeacup** [comment]
I'll do it for you, cashapp me $100

**Comment by regalialong** [comment]
![image](https://github.com/user-attachments/assets/b9ef6113-d8b5-41b7-900a-4bf2434f529f)

**Comment by quazi00** [comment]
> > > how is this abuse? I am not talking to mass block all users in a room, im talking about transfering my own block lists. Nothiung u've staied or these other guys said remotely make any sense.
> > 
> > 
> > applying moderation's against any users affects their rank such as blocking and muting, VRChat is very much against this and any sort of automated moderation's can be abused or used in a targeted manor (KOS Lists) thus why VRChat has expressed a dislike in the idea in the past.
> 
> I could simply reblock everyone, that would also affect their rank. But why when it can be achieved in a better way. If I really wanna hurt a user, I make 100 accounts, and block on all of these accounts a certain user. But who would do that>?! What u say makes 0 sense again.

so you're telling me that if someone made a list of hundreds of known pedos, crashers and racists you wouldn't think "Wow I should import that and keep myself away from all these people." so what happens when even 30% of the VRCX users use this to block all those people? millions of API calls from vrcx (and their API already looks for blocking users in short succession when you could target crash someone with a client by blocking everyone in the lobby but your target and change into the crasher then out of it before unblocking everyone so this is an 11/10 way to get banned lmao.) also "who would do that" You must not have ever met a client user before. Their ultimate goal is to go around crashing people they don't like (AKA the people on this list that could be spread around) and you get a notification when someone who's blocked or muted joins so it's implementing the KOS list without any other work. you would just upload this list of mutes and you don't even have to unblock them to crash them. the fact that you know nothing about vrcx the API or even the player base and are so adamant about how something is "the most important feature" is pretty crazy. if you really want this here are some options. 

1. fork vrcx and make it yourself. you can use chatGTP for a lot these days and it might not take as long as you think to do something like this just use the vrcx source code from here on GitHub and the API documentation on https://vrchatapi.github.io

2. pay someone to do it. things like Fiver igist for a reason or I'm sure just about any client creator will make it for you if you pay them enough. ( you better run every anti-virus on the planet lmao this is a joke)

3. make a discord or GitHub repo where you put all of the User IDs of anyone that you've blocked/muted and share that or keep it for your next account? Idk why this is useful anyway unless you make new accounts or share the list but.

4. use the (currently broken URI (u r i) links such as vrcx://user/usr_ID to make some sort of script to do it for you in autohotkey or some other macro program (again or pay someone to do this) this is probably your best bet for price or time but if you are trying to share this good luck finding anyone who will run your program that basically needs some sort of access to there account lmao.

TLDR; uhhhhh you don't know what you're talking about, it's bad for your account, bad for vrcx, bad for vrchat, and bad for the community. we don't need EAC update for the API lol. I have given you some options on how to proceed if you wanted to.

**Comment by DubyaDude** [comment]
Just a note, the user was already blocked from the repo, no point in engaging them any further.
